misc/drone-scripts/validate-and-serialize.sh

```diff
@@ -13,33 +13,99 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# datastore name refers to storage backed by base ESX (physical ESX) where
+# ongoing/<build_no> contains the directory named after the currently running
+# CI/ build
 DS='-ds=datastore1'
-unit_test_array=($TEST_URL_ARRAY)
-numServers=${#unit_test_array[@]}
+
+# lock names, to use in lock/unlock
+# build_lock:
+#       Overall lock makes sure only one thread at a time runs
+# check_lock:
+#       Represents a lock to perform cleanup leftover from previous run builds.
+build_lock="build_lock"
+check_lock="check_lock"
+
+# refers current CI build#
 DRONE_BUILD_NUMBER=${DRONE_BUILD_NUMBER:=0}
-prevBuildStatus=`drone build info vmware/docker-volume-vsphere $(( $DRONE_BUILD_NUMBER-$numServers ))`
-outArray=($prevBuildStatus)
 
-# Check if any other build is in the ongoing folder, if present, check if entry is stale.
-# If entry is stale clean it up.
-# If entry is an on going build wait for it.
+# wait time in seconds
+interval=90
 
-prevBuild=`govc datastore.ls $DS docker-volume-vsphere/ongoing|tail -n 1`
+# Decide whether cleanup is needed or not.
+# returns true if the ongoing/build info is stale and the build is not running
+# anymore, or if the user wants to restart the build so the ongoing/build and
+# current build have the same ID
+function is_cleanup_needed {
+    # Retrieve on-going build information
+    ongoingBuildInfo=`drone build info vmware/docker-volume-vsphere $1`
+    buildInfoObj=($ongoingBuildInfo)
 
-while [[ "$prevBuild" != "" ]];
-do
-    prevBuildStatus=`drone build info vmware/docker-volume-vsphere $prevBuild`
-    outArray=($prevBuildStatus)
-    if [[ ${outArray[2]} != *"running"* || ${outArray[2]} == *"running"* && $prevBuild == $DRONE_BUILD_NUMBER ]]
+    # verifies status value OR request is resulted from *Restart* event
+    if [[ ${buildInfoObj[2]} != *"running"* || $1 == $DRONE_BUILD_NUMBER ]]
     then
-        govc datastore.rm $DS docker-volume-vsphere/ongoing/$prevBuild
+        return 0
     else
-        echo "Waiting 2 minutes for previous build $prevBuild to complete";
-        sleep 120;
+        return 1
+    fi
+}
+
+# Util to remove created directory (consumed as locking mechanism or
```

Comment:
well after addressing https://github.com/vmware/docker-volume-vsphere/pull/1573#discussion_r126824716 comment .. function is meant for releasing lock only.

I will fix the inline comment before merging.

Thanks!